python/ray/data/aggregate.py

```diff
@@ -57,49 +57,129 @@ def __init__(self):
         on_fn = _to_on_fn(on)
+
+        def accumulate(a: AggType, r: T):
+            r = on_fn(r)
+            if _is_null(r):
+                if ignore_nulls:
+                    res = a
+                else:
+                    raise ValueError(
+                        f"Encountered a null value {r} when calculating the "
+                        "sum; please give ignore_nulls=True if you want nulls "
+                        "to be ignored.")
+            else:
+                res = a + r
+            return res
+
         super().__init__(
             init=lambda k: 0,
-            accumulate=lambda a, r: a + on_fn(r),
+            accumulate=accumulate,
             merge=lambda a1, a2: a1 + a2,
             name=(f"sum({str(on)})"))
 
 
 class Min(AggregateFn):
     """Defines min aggregation."""
 
-    def __init__(self, on: Optional[AggregateOnT] = None):
+    def __init__(self,
+                 on: Optional[AggregateOnT] = None,
+                 ignore_nulls: bool = True):
         on_fn = _to_on_fn(on)
+
+        def accumulate(a: AggType, r: T) -> AggType:
+            r = on_fn(r)
+            if _is_null(r):
+                if ignore_nulls:
+                    res = a
+                else:
+                    raise ValueError(
+                        f"Encountered a null value {r} when calculating the "
+                        "min; please give ignore_nulls=True if you want nulls "
+                        "to be ignored.")
+            else:
+                res = min(a, r) if a is not None else r
+            return res
+
+        def merge(a1: AggType, a2: AggType) -> AggType:
+            if _is_null(a2):
+                return a1
+            if _is_null(a1):
+                return a2
+            return min(a1, a2)
+
         super().__init__(
             init=lambda k: None,
-            accumulate=(
-                lambda a, r: (on_fn(r) if a is None else min(a, on_fn(r)))),
-            merge=lambda a1, a2: min(a1, a2),
+            accumulate=accumulate,
+            merge=merge,
             name=(f"min({str(on)})"))
 
 
 class Max(AggregateFn):
     """Defines max aggregation."""
 
-    def __init__(self, on: Optional[AggregateOnT] = None):
+    def __init__(self,
+                 on: Optional[AggregateOnT] = None,
+                 ignore_nulls: bool = True):
         on_fn = _to_on_fn(on)
+
+        def accumulate(a: AggType, r: T):
+            r = on_fn(r)
+            if _is_null(r):
+                if ignore_nulls:
+                    res = a
+                else:
+                    raise ValueError(
+                        f"Encountered a null value {r} when calculating the "
+                        "max; please give ignore_nulls=True if you want nulls "
+                        "to be ignored.")
+            else:
+                res = max(a, r) if a is not None else r
+            return res
+
+        def merge(a1: AggType, a2: AggType) -> AggType:
+            if _is_null(a2):
```

Comment:
Do we need to check null here? Isn't AggType always valid?

Reply:
This is actually checking for the init value, `None`,  so these should probably just be explicit `None` checks rather than using `_is_null()` here. I'll make that change.

To answer your question, if a block only consists of nulls, then the accumulate stage will ignore the entire block and the init value, `None`, will be propagated as the agg output of the accumulate stage. Therefore, one or more of the two inputs to the merge stage can be `None`.